hcl/writer.go

```diff
@@ -123,3 +124,83 @@ func (w *Writer) Sync() error {
 	}
 	return nil
 }
+
+// Interpolate replaces the hardcoded resources link
+// with TF interpolation
+func (w *Writer) Interpolate(i map[string]string) error {
+	resources := w.Config["resource"]
+	// we need to isolate each resource
+	// getting each resource is easier to avoid cycle
+	// or interpolaception.
+	// We first loop over resource type (e.g: aws_instance)
+	for rt, resource := range resources.(map[string]map[string]interface{}) {
+		// we loop over a resource (e.g: aws_instance.oDSOj)
+		for name, block := range resource {
+			src := reflect.ValueOf(block)
+
+			// this will store the updated block
+			dest := reflect.New(src.Type()).Elem()
+
+			// walk through the resources to interpolate the good values
+			walk(dest, src, i, name, rt)
+
+			// remove reflect.Value wrapper from dest
+			resources.(map[string]map[string]interface{})[rt][name] = dest.Interface()
+		}
+	}
+	return nil
```

Comment:
Why return an error, if it never will?

Reply:
Yeh it could be removed now as it's using a different logic.